hbase-server/src/main/java/org/apache/hadoop/hbase/coprocessor/RegionCoprocessorEnvironment.java

```diff
@@ -26,12 +27,18 @@
 import org.apache.hadoop.hbase.ServerName;
 import org.apache.hadoop.hbase.client.Connection;
 import org.apache.hadoop.hbase.client.RegionInfo;
+import org.apache.hadoop.hbase.client.Scan;
 import org.apache.hadoop.hbase.metrics.MetricRegistry;
+import org.apache.hadoop.hbase.quotas.OperationQuota;
+import org.apache.hadoop.hbase.quotas.RpcQuotaManager;
+import org.apache.hadoop.hbase.quotas.RpcThrottlingException;
 import org.apache.hadoop.hbase.regionserver.OnlineRegions;
 import org.apache.hadoop.hbase.regionserver.Region;
 import org.apache.yetus.audience.InterfaceAudience;
 import org.apache.yetus.audience.InterfaceStability;
 
+import org.apache.hadoop.hbase.shaded.protobuf.generated.ClientProtos;
```

Comment:
Just to check, which protobufs are we exposing to Coprocessor implementations, is it the shaded or unshaded module? Do we have an ErrorProne check that we always do the right thing?

Reply:
Or -- do we ever? Browsing through the other coprocessor interfaces and base classes, I don't see any protobufs used. Instead, it's all pure Java POJOs.

Should this be `org.apache.hadoop.hbase.client.Action` and that class exposed to `IA.LimitedPrivate` ?

Reply:
Ah, but this Quota stuff is all up in the RegionServer's business, where shaded protobufs are indeed used directly.

I think the question stands, of whether we want to expose the shaded protobufs to our coprocessor APIs, or if we need to keep this as a "pure" interface.

@apurtell @Apache9 @busbey where do you stand on this question?

Reply:
According to the ref guide:
> In HBase 3.0.0, the hbase-protocol module has been purged, the CPEP implementation should use the protos in hbase-protocol-shaded module, and also make use of the shaded protobuf in hbase-thirdparty.

I also think I remember Andrew saying that our protos are LimitedPrivate (but cant find that in the refguid). If you look at MultiRowMutationEndpoint in branch-3, it has access to `org.apache.hadoop.hbase.shaded.protobuf.generated.ClientProtos`

So it seems like it should be ok to expose it in master/branch-3. I think in branch-2 it might be more tricky. MultiRowMutationEndpoint only has access to _unshaded_ protobuf there, so it wouldn't translate into the server internals which use _shaded_.

One option here would be to not expose checkBatchQuota with ClientProtos.Action as an argument. The implementation of that method on the server side actually just iterates the batch to count the reads and writes, then calls a different checkBatchQuota that takes those two ints.  We could expose something like `checkBatchQuota(region, int reads, int writes)`.

Reply:
I'm okay with this approach, assuming that we expect downstream implementations to be okay with the lower fidelity summary of the actions list.

Reply:
I've made that change.